src/precompile.jl

```diff
@@ -4,141 +4,195 @@ using Downloads
-                try Base.rm(tmp; recursive=true)
-                catch
+            withenv("JULIA_PKG_PRECOMPILE_AUTO" => 0) do
+                Pkg.add("TestPkg")
+                Pkg.develop(Pkg.PackageSpec(path = "TestPkg.jl"))
+                Pkg.add(Pkg.PackageSpec(path = "TestPkg.jl/"))
+                Pkg.REPLMode.try_prompt_pkg_add(Symbol[:notapackage])
+                Pkg.update(; update_registry = false)
+                Pkg.status()
+                pkgs_path = pkgdir(Pkg, "test", "test_packages")
+                # Precompile a diverse set of test packages
+                # Check all test packages occasionally if anything has been missed
+                # test_packages = readdir(pkgs_path)
+                test_packages = (
+                    "ActiveProjectInTestSubgraph",
+                    "BasicSandbox",
+                    "DependsOnExample",
+                    "PackageWithDependency",
+                    "SameNameDifferentUUID",
+                    "SimplePackage",
+                    "BasicCompat",
+                    "PackageWithDependency",
+                    "SameNameDifferentUUID",
+                    "SimplePackage",
+                    joinpath("ExtensionExamples", "HasExtensions.jl"),
+                )
+                for test_package in test_packages
+                    Pkg.activate(joinpath(pkgs_path, test_package))
                 end
+                Pkg.activate(; temp = true)
+                Pkg.activate()
+                Pkg.activate("TestPkg.jl")
+            end
+            Pkg.precompile()
+            try
+                Base.rm(tmp; recursive = true)
+            catch
+            end
 
-                Base.precompile(Tuple{typeof(Pkg.REPLMode.promptf)})
-                Base.precompile(Tuple{typeof(Pkg.REPLMode.repl_init), REPL.LineEditREPL})
-                Base.precompile(Tuple{typeof(Pkg.API.status)})
-                Base.precompile(Tuple{typeof(Pkg.Types.read_project_compat), Base.Dict{String, Any}, Pkg.Types.Project})
-                Base.precompile(Tuple{typeof(Pkg.Versions.semver_interval), Base.RegexMatch})
-                Base.precompile(Tuple{typeof(REPL.LineEdit.complete_line), Pkg.REPLMode.PkgCompletionProvider, REPL.LineEdit.PromptState})
-                Base.precompile(Tuple{typeof(Pkg.REPLMode.complete_argument), Pkg.REPLMode.CommandSpec, Array{String, 1}, String, Int64, Int64})
-                Base.precompile(Tuple{typeof(Pkg.REPLMode.complete_add_dev), Base.Dict{Symbol, Any}, String, Int64, Int64})
+            Base.precompile(Tuple{typeof(Pkg.REPLMode.promptf)})
+            Base.precompile(Tuple{typeof(Pkg.REPLMode.repl_init),REPL.LineEditREPL})
+            Base.precompile(Tuple{typeof(Pkg.API.status)})
+            Base.precompile(
+                Tuple{
+                    typeof(Pkg.Types.read_project_compat),
+                    Base.Dict{String,Any},
+                    Pkg.Types.Project,
+                },
+            )
```

Comment:
Lets leave these as one-liners.

Reply:
Would you prefer that I do these manually or that I figure out a house style for Pkg.jl?

Reply:
Just manually. I don't think we should go beyond this file